slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/ValidatorsDao.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.slashingprotection;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Supplier;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes;
+
+public class ValidatorsDao {
+  private static final Logger LOG = LogManager.getLogger();
+  private static final String SELECT_VALIDATOR =
+      "SELECT id, public_key FROM validators WHERE public_key IN (%s)";
+  private static final String INSERT_VALIDATOR = "INSERT INTO validators (public_key) VALUES (?)";
+  private final Supplier<Connection> connectionSupplier;
+
+  public ValidatorsDao(final Supplier<Connection> connectionSupplier) {
+    this.connectionSupplier = connectionSupplier;
+  }
+
+  public void registerValidators(final List<Bytes> validators) {
+    final Connection connection = connectionSupplier.get();
+    try {
+      connection.setAutoCommit(false);
+      final List<Bytes> missingValidators = retrieveMissingValidators(connection, validators);
```

Comment:
missingvalidators implies that the validator is in our list but not in db? [can it be named more clearly?]
Assumingly, the other way doesn't matter (i.e. in table, but not in validators?)